### ISSUE TITLE
Ingress Template Generation Issue

### ISSUE THREAD

**Issue by networkhorse** [bootc/netbox-chart#37]
Whilst playing with this in AWS using ALB, I was trying to enforce SSL redirection directly in the ingress configuration.

I looked at some changes and cannot see any obvious issue.

Below are the two types of configuration I have tried:

```
ingress:
  enabled: true
  annotations:
    kubernetes.io/ingress.class: alb
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/target-type: ip
    alb.ingress.kubernetes.io/certificate-arn: redacted
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTP":80},{"HTTPS":443}]'
    alb.ingress.kubernetes.io/actions.ssl-redirect: '{"Type":"redirect","RedirectConfig":{"Protocol":"HTTPS","Port":"443","StatusCode":"HTTP_301"}}'
  hosts:
    - host: redacted
      paths:
        - path: /*
          serviceName: ssl-redirect
          servicePort: use-annotation
        - path: /*
```

and also

```
ingress:
  enabled: true
  annotations:
    kubernetes.io/ingress.class: alb
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/target-type: ip
    alb.ingress.kubernetes.io/certificate-arn: redacted
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTP":80},{"HTTPS":443}]'
    alb.ingress.kubernetes.io/actions.ssl-redirect: '{"Type":"redirect","RedirectConfig":{"Protocol":"HTTPS","Port":"443","StatusCode":"HTTP_301"}}'
  hosts:
    - host: redacted
      paths:
        - path: /*
          backend:
            serviceName: ssl-redirect
            servicePort: use-annotation
        - path: /*
```

Both generate similar configuration to the following:

```
# Source: netbox/templates/ingress.yaml
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: prod-netbox
  labels:
    helm.sh/chart: netbox-2.3.0
    app.kubernetes.io/name: netbox
    app.kubernetes.io/instance: prod
    app.kubernetes.io/version: "v2.9.3"
    app.kubernetes.io/managed-by: Helm
  annotations:
    alb.ingress.kubernetes.io/actions.ssl-redirect: '{"Type":"redirect","RedirectConfig":{"Protocol":"HTTPS","Port":"443","StatusCode":"HTTP_301"}}'
    alb.ingress.kubernetes.io/certificate-arn: redacted
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTP":80},{"HTTPS":443}]'
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/target-type: ip
    kubernetes.io/ingress.class: alb
spec:
  rules:
    - host: "redacted"
      http:
        paths:
          - path: map[backend:map[serviceName:ssl-redirect servicePort:use-annotation] path:/*]
            backend:
              serviceName: prod-netbox
              servicePort: http
          - path: map[path:/*]
            backend:
              serviceName: prod-netbox
              servicePort: http
```

This issue seems to be present in v2.1.0, v2.2.0, and v2.3.0.

It's possible I am doing something wrong, but it's a little tricky finding documentation for this. I had looked in the template here:
https://github.com/bootc/netbox-chart/blob/master/templates/ingress.yaml

...but that didn't seem to provide any clues.

My thought is that this line might be somehow returning true:
```{{- if kindIs "string" . }}```

I am not familiar enough with Helm to suggest a fix.

**Comment by bootc** [comment]
Unfortunately this functionality is not yet in any released version of the chart; it was added after I released 2.3.0. It will be present in the 3.0.0 release later today.

**Comment by networkhorse** [comment]
> Unfortunately this functionality is not yet in any released version of the chart; it was added after I released 2.3.0. It will be present in the 3.0.0 release later today.

Oh fantastic! I'll be first to try it!
